src/main/java/nextstep/security/util/MvcRequestMatcher.java

```diff
@@ -0,0 +1,20 @@
+package nextstep.security.util;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpMethod;
+
+public class MvcRequestMatcher implements RequestMatcher {
+
+    private final HttpMethod method;
+    private final String pattern;
+
+    public MvcRequestMatcher(HttpMethod method, String pattern) {
+        this.method = method;
+        this.pattern = pattern;
+    }
+
+    @Override
+    public boolean matches(HttpServletRequest request) {
+        return this.method.equals(HttpMethod.valueOf(request.getMethod())) && request.getRequestURI().equals(pattern);
```

Comment:
```suggestion
        return this.method.equals(HttpMethod.valueOf(request.getMethod())) && pattern.equals(request.getRequestURI());
```
null-safe 관점에서는 파라미터로 들어오는 `request.getRequestURI()`는 null일 수 있기 때문에 별도로 null 체크를 하지 않는 이상 위와 같이 표현하는 것이 npe가 발생하지 않는 안전한 코드를 만들 수 있습니다.

Reply:
간과하고 있었네요. 수정해두겠습니다! 👍